docs/developer/protocol/coreliquid.md

```diff
@@ -0,0 +1,312 @@
+| 0x02-0x09 | Radiator fan 1 config |
+| 0x0A-0x11 | Radiator fan 2 config |
+| 0x12-0x19 | Radiator fan 3 config |
+| 0x01A-0x21 | Pump speed config |
+| 0x22-0x29 | Waterblock fan config |
+
+A fan temperature config consists of 8 bit integer values:
+  - Mode index
+  - 7 temperature points in Celsius.
+
+### Fan speed config - `0x32` (Get) or `0x40` (Set)
+
+Format:
+
+| Byte index | Description |
+| ---------- | ----------- |
+|  0x00 | 0xD0 |
+| 0x01 | 0x32/0x40 (response/write) |
+| 0x02-0x09 | Radiator fan 1 config |
+| 0x0A-0x11 | Radiator fan 2 config |
+| 0x12-0x19 | Radiator fan 3 config |
+| 0x01A-0x21 | Pump speed config |
+| 0x22-0x29 | Waterblock fan config |
+
+A fan config consists of 8 bit integer values:
+  - Mode index
+  - 7 duty cycle percentage values.
+
+### Get current fan status -  `0x31`
+
+Response:
+
+| Byte index | Description |
+| ---------- | ----------- |
+|  0x00 | 0xD0 |
+| 0x01 | 0x31 |
+| 0x02-0x03 | Radiator fan 1 rpm |
+| 0x04-0x05 | Radiator fan 2 rpm |
+| 0x06-0x07 | Radiator fan 3 rpm |
+| 0x08-0x09 | Pump speed rpm |
+| 0x0A-0x0B | Waterblock fan rpm |
+| 0x16-0x17 | Radiator fan 1 duty % |
+| 0x18-0x19 | Radiator fan 2 duty % |
+| 0x1A-0x1B | Radiator fan 3 duty % |
+| 0x1C-0x1D | Pump speed duty % |
+| 0x1E-0x01F | Waterblock fan duty % |
+
+## Display control
+
+### Show Hardware Monitor - `0x71`
+
+The device is capable of displaying a maximum of 3 different parameters, which will cycle on the display.
+
+| Byte index | Description |
+| ---------- | ----------- |
+|  0x01 | 0x71 |
+|  0x02 | Show CPU frequency (0 or 1) |
+|  0x03 | Show CPU temperature (0 or 1) |
+|  0x04 | Show GPU memory frequency (0 or 1) |
+|  0x05 | Show GPU usage (0 or 1) |
+|  0x06 | Show pump (0 or 1)|
+|  0x07 | Show radiator fan (0 or 1) |
+|  0x08 | Show waterblock fan (0 or 1) |
+|  0x09 | How many radiator fan speeds to show separately (1 or 3) |
+
+### Set User Message - `0x93`
+
+| Byte index | Description |
+| ---------- | ----------- |
+|  0x01 | 0x93 |
+|  0x02-0x3E | Message bytes (ASCII) |
+| 0x3F | 0x20 |
+
+### Set Clock Display - `0x7A`
+
+Sets the clock display style on the OLED screen. `clock_style` determines the visual style of the clock.
+
+| Byte index | Description |
+| ---------- | ----------- |
+|  0x01 | 0x7A |
+|  0x02 | `clock_style` (0, 1 or 2) |
+
+### Set Brightness and Direction - `0x7E`
+
+| Byte index | Description |
+| ---------- | ----------- |
+|  0x01 | 0x7E |
+|  0x02 | brightness (0-100) |
+|  0x03 | direction (0-3) |
+
+## Image Upload Commands
+
+### Upload Image - `0xC0` (GIF) or `0xD0` (Banner)
+
+File uploads are initiated by a single report, after which the data is transferred in chunks of 60 bytes. Uploaded images must be 240x320 px size, and in the standard 24-bit color BMP format. A short sleep (2 seconds has proven safe) should be pla2 seconds has proven safeced between the transfer initiation and start of the data transfer to make sure the device is ready.
```

Comment:
`A short sleep (2 seconds has proven safe) should be pla2 seconds has proven safeced` -> `A short sleep (2 seconds has proven safe)`